File: go/media_models.go
package buble

import (
	"context"
	"net/http"
)

// MediaModelsService provides media model discovery methods.
type MediaModelsService struct {
	client *Client
}

// List returns API-ready media models, modes, inputs, and parameters.
func (s *MediaModelsService) List(ctx context.Context, mediaType string, options ...RequestOption) (*Envelope[[]MediaModel], error) {
	if mediaType != "" {
		options = append(options, WithQuery("media_type", mediaType))
	}
	var out Envelope[[]MediaModel]
	if err := s.client.do(ctx, http.MethodGet, "/api/v1/media_models", nil, &out, options...); err != nil {
		return nil, err
	}
	return &out, nil
}

File: rust/src/media_models.rs
use reqwest::Method;

use crate::{client::Client, error::Result, Envelope, MediaModel};

/// Media model discovery methods.
#[derive(Clone, Debug)]
pub struct MediaModelsService {
    client: Client,
}

impl MediaModelsService {
    pub(crate) fn new(client: Client) -> Self {
        Self { client }
    }

    /// Lists API-ready media models.
    pub async fn list(&self, media_type: Option<&str>) -> Result<Envelope<Vec<MediaModel>>> {
        let query;
        let query_ref = if let Some(media_type) = media_type {
            query = [("media_type", media_type.to_string())];
            Some(query.as_slice())
        } else {
            None
        };
        self.client
            .request_json::<Envelope<Vec<MediaModel>>, ()>(
                Method::GET,
                "/api/v1/media_models",
                query_ref,
                None,
            )
            .await
    }
}
